sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TableOutputResolver.scala

```diff
@@ -455,6 +457,19 @@ object TableOutputResolver {
     }
   }
 
+  def suitableForByNameCheck(
+      byName: Boolean,
+      expected: Seq[Attribute],
+      queryOutput: Seq[Attribute]): Unit = {
+    if (!byName && expected.size == queryOutput.size &&
+      expected.forall(e => queryOutput.exists(p => conf.resolver(p.name, e.name))) &&
+      expected.zip(queryOutput).exists(e => !conf.resolver(e._1.name, e._2.name))) {
+      logWarning("The query columns and the table columns have same names but different " +
+        "orders. You can use INSERT [INTO | OVERWRITE] BY NAME to reorder the query columns to " +
+        "align with the table columns.")
```

Comment:
Users always insert table columns doesn't match the query columns. For example, table `student(name: STRING, address: STRING)` and table `location(code: STRING, name: STRING)`. The `student.address` references `location.name`.
I am neutral on this warning message.

Reply:
I know that. We can extend the example above to satisfy the condition you mentioned.

Reply:
Right but it's a warning message so if a user is doing it intentionally they can ignore the warning.